router/batchrouter/asyncdestinationmanager/eloqua/bulk_uploader.go

```diff
@@ -0,0 +1,227 @@
+package eloqua
+
+import (
+	stdjson "encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+)
+
+func createAsyncUploadErrorOutput(errorString string, err error, destinationId string, asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	return common.AsyncUploadOutput{
+		FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+		FailedReason:  fmt.Sprintf(errorString+"%v", err),
+		FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+		DestinationID: destinationId,
+	}
+}
+
+func (b *EloquaBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	destination := asyncDestStruct.Destination
+	file, err := os.Open(asyncDestStruct.FileName)
+	if err != nil {
+		return createAsyncUploadErrorOutput("got error while opening the file. ", err, destination.ID, asyncDestStruct)
+	}
+	defer file.Close()
+	eventDetails, err := getEventDetails(file)
+	if err != nil {
+		return createAsyncUploadErrorOutput("got error while checking the event type. ", err, destination.ID, asyncDestStruct)
+	}
+
+	customObjectData := HttpRequestData{
+		BaseEndpoint:  b.baseEndpoint,
+		Authorization: b.authorization,
+	}
+	if eventDetails.Type == "track" {
+		customObjectData.DynamicPart = eventDetails.CustomObjectId
+	}
+	eloquaFields, err := b.service.FetchFields(&customObjectData)
+	if err != nil {
+		return createAsyncUploadErrorOutput("got error while fetching fields. ", err, destination.ID, asyncDestStruct)
+	}
+
+	uploadJobInfo := JobInfo{
+		fileSizeLimit: b.fileSizeLimit,
+		importingJobs: asyncDestStruct.ImportingJobIDs,
+	}
+
+	importDefinitionBody, err := createBodyForImportDefinition(eventDetails.Type, eventDetails.Fields, eloquaFields, eventDetails.IdentifierFieldName)
```

Comment:
createBodyForImportDefinition(eventDetails, eloquaFields) less params will be better